client/src/planwise/client/scenarios/handlers.cljs

```diff
@@ -296,89 +322,95 @@
  :scenarios.new-action/toggle-options
```

Comment:
This handler is doing too much. At the very least I'd try to separate the logic for cancelling the ongoing computations from opening/closing the options menu. Of course you'll need to change the view code as well, but that should be straightforward. Having done that, this should be trivial to implement: if the view state is `show-options-to-create-provider`, then pop the view stack. Otherwise, push the `show-options-to-create-provider` to the stack. For cancelling the computations there already is an existing handler `abort-fetching-suggestions`  but that requires an extra argument. You can either dispatch that event from the view directly (with the appropriate argument), or change the handler to find which of the computations is active.